src/hooks/useApproveAndDeposit.ts

```diff
@@ -60,18 +60,20 @@ export function useApproveAndDeposit(
     transactionDeadlineSelected,
     infiniteApproval,
   } = useSelector((state: AppState) => state.user)
-  const POOL = POOLS_MAP[poolName]
+  const basicPools = useContext(BasicPoolsContext)
+  const tokens = useContext(TokensContext)
+  const pool = basicPools?.[poolName]
   const metaSwapContract = useMemo(() => {
-    if (POOL.metaSwapAddresses && chainId && library) {
+    if (pool?.poolAddress && chainId && library) {
```

Comment:
@hammeiam can you confirm this naming conversion is correct? I read the comment about it but want to make sure.